drivers/unix/os_unix.cpp

```diff
@@ -751,12 +751,23 @@ String OS_Unix::get_executable_path() const {
 		return OS::get_executable_path();
 	}
 	return b;
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__)
 	char resolved_path[MAXPATHLEN];
 
 	realpath(OS::get_executable_path().utf8().get_data(), resolved_path);
 
 	return String(resolved_path);
+#elif defined(__NetBSD__)
+	int mib[4] = { CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME };
+	char buf[MAXPATHLEN];
+	size_t len = sizeof(buf);
+	if (sysctl(mib, 4, buf, &len, nullptr, 0) != 0) {
+		WARN_PRINT("Couldn't get executable path from sysctl");
+		return OS::get_executable_path();
+	}
+	String b;
+	b.parse_utf8(buf);
+	return b;
```

Comment:
Yeah I was wondering how you felt about that. I reviewed former code and it appeared you guys preferred having separate copies of code for each platform, but I am more than happy to remove the redundancy. I wanted to do that initially as my personal preference, but I wasn't sure what everyone else's thought was on the matter. 

Thank you for the code review. I will make that change now.

Reply:
In regards to structure, DragonFly BSD uses the exact same code as as FreeBSD 1:1 without a single line touched. OpenBSD doesn't even have this feature implemented OS-level by contrast.

Reply:
I can revert my commit I just made if everyone decides to stick with my original code change that kept the duplicate code, though, I wouldn't recommend it. I prefer what you suggested.

Reply:
Additionally, there is a bug with NetBSD that if argv[0] is a relative path (for example: `./a.out`) it will return an absolute path by KERN_PROC_PATHNAME sysctl(), though in my experience, it keeps the "relative dots" in the string. For example, if argv[0] is `./a.out`, KERN_PROC_PATHNAME will give you a path looking line `/home/netbsd/./a.out` for example. So, my proposal is to clean up after this NetBSD bug and normalize the pathname with a call to `realpath`. Just my thought on the matter.

Reply:
No problem! Sure thing.